packages/opentelemetry-resources/src/platform/node/ProcessDetector.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { diag } from '@opentelemetry/api';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { Resource } from '../../Resource';
+import { Detector, ResourceAttributes } from '../../types';
+import { ResourceDetectionConfig } from '../../config';
+import * as os from 'os';
+
+/**
+ * ProcessDetector will be used to detect the resources related current process running
+ * and being instrumented from the NodeJS Process module.
+ */
+class ProcessDetector implements Detector {
+  async detect(_config?: ResourceDetectionConfig): Promise<Resource> {
+    const attributes: ResourceAttributes = {
+      [SemanticResourceAttributes.PROCESS_PID]: process.pid,
+      [SemanticResourceAttributes.PROCESS_EXECUTABLE_NAME]: process.title,
+      [SemanticResourceAttributes.PROCESS_EXECUTABLE_PATH]: process.execPath,
+      [SemanticResourceAttributes.PROCESS_COMMAND_LINE]: process.argv.join(' '),
+      [SemanticResourceAttributes.PROCESS_COMMAND_ARGS]: process.argv,
+      [SemanticResourceAttributes.PROCESS_RUNTIME_VERSION]:
+        process.versions.node,
+      [SemanticResourceAttributes.PROCESS_RUNTIME_NAME]: 'nodejs',
+      [SemanticResourceAttributes.PROCESS_RUNTIME_DESCRIPTION]: 'Node.js',
+    };
+
+    if (process.argv.length > 1) {
+      attributes[SemanticResourceAttributes.PROCESS_COMMAND] = process.argv[1];
```

Comment:
This one (might) be dangerous, as (if) an application is passing in some form of credentials (or PII) on the command line this will cause them to get "logged"...

It might be worth having some sort of configuration or something to block this from getting logged (or require an explicit configuration to cause it to get logged), so that this can't inadvertently occur.

Reply:
This processor is already logging `[SemanticResourceAttributes.PROCESS_COMMAND_LINE]: process.argv.join(' '),` without configuration. I agree that being able to disable collection isn't a bad idea, but pointing out this is not a change in behavior being introduced in this PR.

Reply:
As previously mentioned this Processor is already collecting `process.argv[1]`, twice in fact. See: https://github.com/open-telemetry/opentelemetry-js/blob/47444f2a4c13a80f3328ffafc5b4c8db9e9a0886/packages/opentelemetry-resources/src/detectors/ProcessDetectorSync.ts#L37-L39.

We have a couple of options forward.

* We could leave things as is. This processor is not a default processor, and users need to opt-in to collecting data with it.
* We could add additional configuration. It would have to go into [ResourceDetectionConfig](https://github.com/open-telemetry/opentelemetry-js/blob/47444f2a4c13a80f3328ffafc5b4c8db9e9a0886/packages/opentelemetry-resources/src/config.ts#L22-L24) which is shared between all detectors. We would need to decide on a name, and if collection was turned off, it would probably omit all of the `process.argv` based attributes in this detector.
* We could do a combination, leave this as-is, since this doesn't change the current behavior, and make an issue for optional collection to be addressed later.

If anyone has strong opinions let me know. I'm happy to address this is anyway we see fit.

Reply:
As discussed in the SIG, here we are just implementing the spec. If there is a problem with the spec, it should be addressed there.